realm/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -294,29 +323,48 @@ public void testGetObjects() {
         assertEquals("Dog 1", dogs.get(1).getName());
     }
 
-    public void testGetFirstObject() {
+    public void testFirst() {
         Owner owner = testRealm.where(Owner.class).findFirst();
         Dog dog = owner.getDogs().first();
 
         assertEquals("Dog 0", dog.getName());
     }
 
-    public void testGetLastObject() {
+    public void testFirstEmptyList() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        testRealm.beginTransaction();
+        owner.getDogs().clear();
+        testRealm.commitTransaction();
+
+        assertNull(owner.getDogs().first());
+    }
+
+    public void testLast() {
         Owner owner = testRealm.where(Owner.class).findFirst();
         Dog dog = owner.getDogs().last();
 
         assertEquals("Dog " + (TEST_OBJECTS - 1), dog.getName());
     }
 
+    public void testLastEmptyList() {
```

Comment:
merge with testFirstEmptyList. 
Too much duplicate code. Same as above comment, but here the benefits are even more clear :-)